### PR TITLE
Fix coin change

### PR DESCRIPTION
**Pull request by himadriganguly**
### **Describe your change:**
Removed unused variable and modified code for coin_chainge algo.

* [ ] Add an algorithm?
* [X] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [X] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [X] This pull request is all my own work -- I have not plagiarized.
* [X] I know that pull requests will not be merged if they fail the automated tests.
* [X] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [X] All new Python files are placed inside an existing directory.
* [X] All filenames are in all lowercase characters with no spaces or dashes.
* [X] All functions and variable names follow Python naming conventions.
* [ ] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Review by Kush1101 — COMMENTED** [review]
Is it really a useful contribution?

File: dynamic_programming/coin_change.py [hunk 1]
```diff
@@ -7,20 +7,23 @@
...
     0
-    >>> dp_count([4, 5, 6], 3, 0)
+    >>> dp_count([4, 5, 6], 0)
     1
+    >>> dp_count([1, 2, 3], -5)
+    0
     """
-
+    if n < 0:
+        return(0)
```
Comment (poyea) [inline dynamic_programming/coin_change.py:26]: Can you cover this by adding a few test cases?
```suggestion
    if n < 0:
        return 0
```

**Review by poyea — CHANGES_REQUESTED** [review]

File: dynamic_programming/coin_change.py (hunk 1 above)
Comment (himadriganguly) [reply dynamic_programming/coin_change.py:26]: @poyea Now the test is passing in my local system. Also added 4 test case.

File: dynamic_programming/coin_change.py [hunk 2]
```diff
@@ -7,20 +7,29 @@
...
+    >>> dp_count([4, 5, 6], 0)
     1
+    >>> dp_count([1, 2, 3], -5)
+    0
+    >>> dp_count([4, 5, 6], -4)
+    0
+    >>> dp_count([2, 5, 3, 6], -10)
+    0
+    >>> dp_count([4], -4)
+    0
```
Comment (poyea) [inline dynamic_programming/coin_change.py:29]:
```suggestion
    >>> dp_count([1, 2, 3], -5)
    0
```
I think 1 test case for this is enough.

**Comment by poyea** [comment]
Can you change the title of the pull request to something like `fix coin change`?

File: dynamic_programming/coin_change.py (hunk 2 above)
Comment (himadriganguly) [reply dynamic_programming/coin_change.py:29]: @poyea You said to add some more test cases in the previous comment?

Comment (poyea) [reply dynamic_programming/coin_change.py:29]: I mean to test the `< 0` line, one test case is enough. You may add some larger test cases too to demonstrate the correctness.

**Comment by himadriganguly** [comment]
@poyea done the changes have a look.

**Review by poyea — APPROVED** [review]

**Pull request merged**